internal/controller/pod_webhook.go

```diff
@@ -121,3 +125,33 @@ func hasMIGResource(pod *v1.Pod) bool {
 	}
 	return false
 }
+
+func performQuotaArithmetic(pod *v1.Pod, req admission.Request) admission.Response {
+	// assumption is that workloads will have 1 container where
+	// MIG is requested.
+	for _, container := range pod.Spec.Containers {
+		// dont bother checking requests section. Nvidia supports only limits
+		// if requests is added by user, it should be equal to limits.
+		for resourceName, quantity := range container.Resources.Limits {
+			resourceParts := strings.Split(strings.TrimPrefix(string(resourceName), "nvidia.com/mig-"), ".")
+
+			if len(resourceParts) == 2 {
+				//gpuPart := resourceParts[0]
+				memoryPart := resourceParts[1]
+				memoryValue, err := strconv.Atoi(strings.TrimSuffix(memoryPart, "gb"))
+				if err != nil {
+					return admission.Errored(http.StatusInternalServerError, fmt.Errorf("failed to parse memory value: %v", err))
+				}
+				acceleratorMemory := memoryValue * int(quantity.Value())
+				// assume 1 container workload
+				pod.Spec.Containers[0].Resources.Limits["nvidia.com/accelerator-memory"] = resource.MustParse(fmt.Sprintf("%dGi", acceleratorMemory))
```

Comment:
`nvidia.com/accelerator-memory` more than one occurrence, maybe a good fit for string constant?